### PR TITLE
Bug fix, some CI updates and codespace badge

### PR DESCRIPTION
**Pull request by OnnoEbbens**

### CODE REVIEW

File: hydropandas/io/bro.py [hunk 1]
```diff
@@ -422,8 +423,9 @@ def get_full_metadata_from_gmw(bro_id, tube_nr):
     return meta
 
 
-def get_tube_nrs_from_gmw(bro_id):
-    """returns all tube numbers from a groundwater monitoring well (gmw)
+@lru_cache()
+def _get_gmw_from_bro_id(bro_id, retries=0):
```
Comment (dbrakenhoff) [inline hydropandas/io/bro.py:427]: is retries number of retries or a counter that goes up to 5?

File: hydropandas/io/bro.py [hunk 2]
```diff
@@ -432,31 +434,70 @@ def get_tube_nrs_from_gmw(bro_id):
...
 
     # read results
     tree = xml.etree.ElementTree.fromstring(req.text)
 
-    # get gmw
     gmws = tree.findall(".//dsgmw:GMW_PO", ns)
     if len(gmws) != 1:
+        val_ind = req.text.find("valid")
+        valid = req.text[(val_ind + 9) : (val_ind + 14)]
+        if valid == "false" and retries < 5:
```
Comment (dbrakenhoff) [inline hydropandas/io/bro.py:465]: maybe allow configuring of max_retries, instead of hard-coding?

File: hydropandas/io/bro.py [hunk 3]
```diff
@@ -432,31 +434,70 @@ def get_tube_nrs_from_gmw(bro_id):
...
+            logger.debug(
+                f"got invalid response for {bro_id}, trying again {retries+1}/4"
```
Comment (dbrakenhoff) [inline hydropandas/io/bro.py:467]: see above.

File: hydropandas/io/bro.py (hunk 1 above)
Comment (dbrakenhoff) [reply hydropandas/io/bro.py:427]: I see it's recursive now. So nvm, but maybe good to allow max_retries to be configurable?

Comment (OnnoEbbens) [reply hydropandas/io/bro.py:428]: Good idea, I've added rcParams for this

**Review by dbrakenhoff — DISMISSED** [review]
Looks good!

**Review by ArtesiaWater — APPROVED** [review]

**Pull request merged**